src/main.ts

```diff
@@ -3,56 +3,58 @@ import fs from 'fs'
 import glob from 'glob'
 import chromeWebstoreUpload from 'chrome-webstore-upload'
 
-function uploadFile(
+async function uploadFile(
   webStore: any,
   filePath: string,
-  publishFlg: string,
+  publishFlg: 'true' | 'false',
   publishTarget: string
-): void {
+): Promise<void> {
   const myZipFile = fs.createReadStream(filePath)
-  webStore
-    .uploadExisting(myZipFile)
-    .then((uploadRes: any) => {
-      console.log(uploadRes)
-      core.debug(uploadRes)
 
-      if (
-        uploadRes.uploadState &&
-        (uploadRes.uploadState === 'FAILURE' ||
-          uploadRes.uploadState === 'NOT_FOUND')
-      ) {
-        uploadRes.itemError.forEach((itemError: any) => {
-          core.error(
-            Error(`${itemError.error_detail} (${itemError.error_code})`)
-          )
-        })
-        core.setFailed(
-          'upload error - You will need to go to the Chrome Web Store Developer Dashboard and upload it manually.'
-        )
-        return
-      }
+  try {
+    const uploadRes = await webStore.uploadExisting(myZipFile)
 
-      if (publishFlg === 'true') {
-        webStore
-          .publish(publishTarget)
-          .then((publishRes: any) => {
-            core.debug(publishRes)
-          })
-          .catch((e: any) => {
-            core.error(e)
-            core.setFailed(
-              'publish error - You will need to access the Chrome Web Store Developer Dashboard and publish manually.'
-            )
-          })
-      }
-    })
-    .catch((e: any) => {
-      console.log(e)
-      core.error(e)
+    console.log(uploadRes)
+    core.debug(uploadRes)
+
+    if (
+      uploadRes.uploadState &&
+      (uploadRes.uploadState === 'FAILURE' ||
+        uploadRes.uploadState === 'NOT_FOUND')
+    ) {
+      uploadRes.itemError.forEach((itemError: any) => {
+        core.error(
+          new Error(`${itemError.error_detail} (${itemError.error_code})`)
+        )
+      })
       core.setFailed(
         'upload error - You will need to go to the Chrome Web Store Developer Dashboard and upload it manually.'
       )
-    })
+      throw new Error(
+        'upload error - You will need to go to the Chrome Web Store Developer Dashboard and upload it manually.'
+      )
```

Comment:
The run function catches any error messages that occur and displays them using the setFailed function. 
As the same message is likely to be displayed, it may be unnecessary to use setFailed in the uploadFile function.